src/material/expansion/testing/accordion-harness.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {MatExpansionPanelHarness} from './expansion-harness';
+import {AccordionHarnessFilters} from './expansion-harness-filters';
+
+/** Harness for interacting with a standard mat-accordion in tests. */
+export class MatAccordionHarness extends ComponentHarness {
+  static hostSelector = '.mat-accordion';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for an accordion
+   * with specific attributes.
+   * @param options Options for narrowing the search.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: AccordionHarnessFilters = {}): HarnessPredicate<MatAccordionHarness> {
+    return new HarnessPredicate(MatAccordionHarness, options);
+  }
+
+  private _expansionPanels = this.locatorForAll(MatExpansionPanelHarness);
+
+  /** Gets all expansion panels which are part of the accordion. */
+  async getExpansionPanels(): Promise<MatExpansionPanelHarness[]> {
```

Comment:
For these methods that expose sub-harnesses, I've been having them take an optional filter object (in this case `ExpansionPanelHarnessFilters`) that way users can easily ask for a particular panel or subset of the panels that match some criteria

Reply:
Agreed. I just forgot about this, but recall that we decided doing this for the tab harness too.